services/clients/firebase_client.py

```diff
@@ -9,48 +9,74 @@
 from google.cloud.firestore_v1.client import Client
 from google.cloud.storage import Blob
 
+
 class FirebaseClient:
-    TEST_COLLECTION = "documentation"
+    DOCUMENTATION_COLLECTION = "documentation"
 
     def __init__(self):
         self.bucket = storage.bucket()
         self.db: Client = firestore.client()
 
-    def add_document(self, collection_name, data) -> DocumentReference:
-        collection_ref = self.db.collection(collection_name)
+    def add_blob(self, blob_url, data):
+        blob = self.bucket.blob(blob_url)
+        blob.upload_from_string(data)
+
+    def get_blob(self, blob_url) -> Blob:
+        return self.bucket.get_blob(blob_url)
+
+    def get_documentation(self, doc_id) -> Dict[str, Any] | None:
+        document_snapshot = self._get(self.DOCUMENTATION_COLLECTION, doc_id)
+        if not document_snapshot:
+            return None
+
+        document_dict = document_snapshot.to_dict()
+        document_dict["id"] = document_snapshot.id
+
+        return document_dict
+
+    def add_documentation(self, data) -> str:
+        document_ref = self._add(
+            FirebaseClient.DOCUMENTATION_COLLECTION,
+            data
+        )
+
+        return document_ref.id
+
+    def update_documentation(self, doc_id: str, data) -> None:
+        self._update(
+            self.DOCUMENTATION_COLLECTION,
+            doc_id,
+            data
+        )
+
+    @staticmethod
+    def get_blob_url(blob_name, folder="repo") -> str:
+        return f"/{folder}/{blob_name}"
+
+    def _add(self, collection_path, data) -> DocumentReference:
+        collection_ref = self.db.collection(collection_path)
         update_time, document_ref = collection_ref.add(data)
         return document_ref
 
-    def get_document(self, collection_name, document_id) -> DocumentSnapshot | None:
-        document_ref = self.db.collection(collection_name).document(document_id)
+    def _get(self, collection_path, document_id) -> DocumentSnapshot | None:
```

Comment:
made these internal, and renamed from `*action*_document`  to simply `_*action*` since it might be confusing considering our objects are called `documentation`. Now we can reuse them freely with any future objects.